### ISSUE TITLE
after routing to another route and going back, pagination shows only 1 page

### ISSUE THREAD

**Issue by masteriusz** [donnelljenkins/aurelia-pagination#4]
After the second activation of all views with aurelia-datagrid and pagination, pagination will show only 1 page.

Example:
start application: `localhost:3000`
click nav-bar link, route to customers `http://localhost:3000/#/customers` <-- Everything works correctly.
click nav-bar link, route to welcome `http://localhost:3000/#`
click nav-bar link, route to  customers `http://localhost:3000/#/customers` <-- Pagination shows only one page.

The problem exists in all my datagrids, both while using applications routing links or just directly typing addresses into web browser. Only refreshing helps.

**Comment by el95149** [comment]
Hello, I faced similar issues. Version 0.1.6 fixed a lot of those.
Additionally, initializing my grid data within the 'activate' function, as follows, helped a lot:
`activate() {
        return this.http.fetch('product')
            .then(response => response.json())
            .then(products => {
                for (var product of products) {
                    this.products.push(product);
                }
            });
`
the important part above is the **return** statement. Without it, my grid wouldn't initialize properly.
Hope I helped, regards,

**Comment by masteriusz** [comment]
Hello,
Funny things happened. After installing another plugin my application died completely, forcing me into digging what was the problem. Eventually, I found out that there were many different versions of the same core aurelia packages (like templating, path, pal, so on.. a bunch of them).
**Solution:**
After doing `jspm inspect` and `jspm resolve --only` with each package, not only did I revive the app, but also the pagination problem is gone. 
Summing up - didn't even have to try your solution, sometimes misfortune turns into luck.
